configstack/running_module.go

```diff
@@ -184,13 +202,23 @@ func (module *runningModule) runNow() error {
 	}
 }
 
+var separator = strings.Repeat("-", 132)
+
 // Record that a module has finished executing and notify all of this module's dependencies
 func (module *runningModule) moduleFinished(moduleErr error) {
-	if moduleErr == nil {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished successfully!", module.Module.Path)
-	} else {
-		module.Module.TerragruntOptions.Logger.Printf("Module %s has finished with an error: %v", module.Module.Path, moduleErr)
+	status := "successfully!"
+	if moduleErr != nil {
+		status = fmt.Sprintf("with an error: %v", moduleErr)
+	}
+
+	module.Mutex.Lock()
+	defer module.Mutex.Unlock()
```

Comment:
Why do you need a lock here? I'd prefer to avoid locks as much as possible and use solely goroutines and channels.